### PR TITLE
Added missed API and improved for Wapo version's Gateway

### PR DESCRIPTION
**Pull request by Leechael**

### CODE REVIEW

File: WapoJS/src/native_runtime.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ pub fn sni_listen(sni: &str, cert: &str, key: &str) -> Result<TlsListener> {
     let listener = global_sni_listener()?;
     let key = sni_tls_listener::wrap_certified_key(cert.as_bytes(), key.as_bytes())
         .context("invalid cert or key")?;
+    log::info!(target: "wapo:tls", "SNI listener is listening on {sni}");
```
Comment (kvinwang) [inline WapoJS/src/native_runtime.rs]:
```suggestion
    log::info!(target: "wapo::tls", "SNI listener is listening on {sni}");
```

File: WapoJS/src/js_eval.rs [hunk 2]
```diff
@@ -56,6 +62,23 @@ fn parse_args(args: impl Iterator<Item = String>) -> Result<Args> {
...
+                    let path_str = iter.next().ok_or(anyhow!("missing path after -e"))?;
+                    let path = std::path::Path::new(&path_str);
+                    if !path.exists() {
+                        return Err(anyhow!("path {path_str} not exists."));
+                    }
+                    if let Err(_) = dotenv::from_path(path) {
+                        return Err(anyhow!("not a valid env file: {path_str}"));
+                    }
+                }
+                "--worker-secret" => {
```
Comment (kvinwang) [inline WapoJS/src/js_eval.rs:77]: Should be `native`-only.

File: WapoJS/src/js_eval.rs [hunk 3]
```diff
@@ -71,25 +94,33 @@ fn parse_args(args: impl Iterator<Item = String>) -> Result<Args> {
         print_usage();
         bail!("no script file provided");
     }
+    if worker_secret.is_none() {
```
Comment (kvinwang) [inline WapoJS/src/js_eval.rs:99]: I feel it inconvenient to use wapojs to do tests if the `--worker-secret` is a required option. Should we give it a default value?

Comment (Leechael) [reply WapoJS/src/js_eval.rs:99]: Since it's only required for native build, how about give a default value and showing a warning message?

Comment (kvinwang) [reply WapoJS/src/js_eval.rs:99]: > Since it's only required for native build, how about give a default value and showing a warning message?

Good idea.

**Review by kvinwang — APPROVED** [review]
Awesome!

**Pull request merged**